examples/multimodal/wds.py

```diff
@@ -14,14 +14,14 @@
     DataChain.from_parquet("gs://datachain-demo/datacomp-small/metadata/0020f*.parquet")
     .settings(cache=True)
     .merge(wds_images, on="uid", right_on="laion.json.uid", inner=True)
-    .mutate(stem=path.file_stem(C("source.file.name")))
+    .mutate(stem=path.file_stem(path.name(C("source.file.path"))))
```

Comment:
It is late here and I could be wrong but it looks like the combination of `path.file_stem` with `path.name` currently generates SQL that looks like this:

```sql
              rtrim(
                substr(
                  ltrim(
                    substr(
                      source__file__path,
                      length(
                        rtrim(
                          rtrim(
                            source__file__path,
                            replace(source__file__path, :param_1, :param_2)
                          ),
                          :param_1
                        )
                      ) + :length_16
                    ),
                    :param_1
                  ),
                  :substr_2,
                  length(
                    ltrim(
                      substr(
                        source__file__path,
                        length(
                          rtrim(
                            rtrim(
                              source__file__path,
                              replace(source__file__path, :param_1, :param_2)
                            ),
                            :param_1
                          )
                        ) + :length_17
                      ),
                      :param_1
                    )
                  ) - CASE WHEN (
                    instr(
                      ltrim(
                        substr(
                          ltrim(
                            substr(
                              source__file__path,
                              length(
                                rtrim(
                                  rtrim(
                                    source__file__path,
                                    replace(source__file__path, :param_1, :param_2)
                                  ),
                                  :param_1
                                )
                              ) + :length_18
                            ),
                            :param_1
                          ),
                          length(
                            rtrim(
                              rtrim(
                                ltrim(
                                  substr(
                                    source__file__path,
                                    length(
                                      rtrim(
                                        rtrim(
                                          source__file__path,
                                          replace(source__file__path, :param_1, :param_2)
                                        ),
                                        :param_1
                                      )
                                    ) + :length_19
                                  ),
                                  :param_1
                                ),
                                replace(
                                  ltrim(
                                    substr(
                                      source__file__path,
                                      length(
                                        rtrim(
                                          rtrim(
                                            source__file__path,
                                            replace(source__file__path, :param_1, :param_2)
                                          ),
                                          :param_1
                                        )
                                      ) + :length_20
                                    ),
                                    :param_1
                                  ),
                                  :param_1,
                                  :param_2
                                )
                              ),
                              :param_1
                            )
                          ) + :length_21
                        ),
                        :param_1
                      ),
                      :param_3
                    ) = :instr_2
                  ) THEN :param_5 ELSE length(
                    ltrim(
                      substr(
                        ltrim(
                          substr(
                            source__file__path,
                            length(
                              rtrim(
                                rtrim(
                                  source__file__path,
                                  replace(source__file__path, :param_1, :param_2)
                                ),
                                :param_1
                              )
                            ) + :length_22
                          ),
                          :param_1
                        ),
                        length(
                          rtrim(
                            rtrim(
                              ltrim(
                                substr(
                                  source__file__path,
                                  length(
                                    rtrim(
                                      rtrim(
                                        source__file__path,
                                        replace(source__file__path, :param_1, :param_2)
                                      ),
                                      :param_1
                                    )
                                  ) + :length_23
                                ),
                                :param_1
                              ),
                              replace(
                                ltrim(
                                  substr(
                                    source__file__path,
                                    length(
                                      rtrim(
                                        rtrim(
                                          source__file__path,
                                          replace(source__file__path, :param_1, :param_2)
                                        ),
                                        :param_1
                                      )
                                    ) + :length_24
                                  ),
                                  :param_1
                                ),
                                :param_1,
                                :param_2
                              )
                            ),
                            :param_1
                          )
                        ) + :length_21
                      ),
                      :param_1
                    )
                  ) - length(
                    rtrim(
                      ltrim(
                        substr(
                          ltrim(
                            substr(
                              source__file__path,
                              length(
                                rtrim(
                                  rtrim(
                                    source__file__path,
                                    replace(source__file__path, :param_1, :param_2)
                                  ),
                                  :param_1
                                )
                              ) + :length_25
                            ),
                            :param_1
                          ),
                          length(
                            rtrim(
                              rtrim(
                                ltrim(
                                  substr(
                                    source__file__path,
                                    length(
                                      rtrim(
                                        rtrim(
                                          source__file__path,
                                          replace(source__file__path, :param_1, :param_2)
                                        ),
                                        :param_1
                                      )
                                    ) + :length_26
                                  ),
                                  :param_1
                                ),
                                replace(
                                  ltrim(
                                    substr(
                                      source__file__path,
                                      length(
                                        rtrim(
                                          rtrim(
                                            source__file__path,
                                            replace(source__file__path, :param_1, :param_2)
                                          ),
                                          :param_1
                                        )
                                      ) + :length_27
                                    ),
                                    :param_1
                                  ),
                                  :param_1,
                                  :param_2
                                )
                              ),
                              :param_1
                            )
                          ) + :length_21
                        ),
                        :param_1
                      ),
                      replace(
                        ltrim(
                          substr(
                            ltrim(
                              substr(
                                source__file__path,
                                length(
                                  rtrim(
                                    rtrim(
                                      source__file__path,
                                      replace(source__file__path, :param_1, :param_2)
                                    ),
                                    :param_1
                                  )
                                ) + :length_28
                              ),
                              :param_1
                            ),
                            length(
                              rtrim(
                                rtrim(
                                  ltrim(
                                    substr(
                                      source__file__path,
                                      length(
                                        rtrim(
                                          rtrim(
                                            source__file__path,
                                            replace(source__file__path, :param_1, :param_2)
                                          ),
                                          :param_1
                                        )
                                      ) + :length_29
                                    ),
                                    :param_1
                                  ),
                                  replace(
                                    ltrim(
                                      substr(
                                        source__file__path,
                                        length(
                                          rtrim(
                                            rtrim(
                                              source__file__path,
                                              replace(source__file__path, :param_1, :param_2)
                                            ),
                                            :param_1
                                          )
                                        ) + :length_30
                                      ),
                                      :param_1
                                    ),
                                    :param_1,
                                    :param_2
                                  )
                                ),
                                :param_1
                              )
                            ) + :length_21
                          ),
                          :param_1
                        ),
                        :param_3,
                        :param_2
                      )
                    )
                  ) END
                ),
                :param_3
              ) AS stem
```

Which seems to be contributing to the `sqlite3.OperationalError: parser stack overflow` errors that I am now seeing for #199 ([example](https://github.com/iterative/datachain/actions/runs/10299269300/job/28506233374#step:5:1350)). Using `.mutate(stem=path.file_stem(C("source.file.path")))` seems to patch the issue. I'll take a look tomorrow to see if I can contribute anything more meaningful.